### PR TITLE
Update TRAVIS in order to test Java tools

### PR DESCRIPTION
**Pull request by mcoolive**
- Specify the version of Python
- Specify the version of the JDK, and install it
- Add execution of existing Java tests. Run once to save time (on the
slave that checks pyjama).

---
This change is [Reviewable](https://reviewable.io/reviews/criteo/biggraphite/188)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,26 +1,44 @@
...
+  apt:
+    packages:
+      - oracle-java8-installer
+
 env:
   matrix:
-    - TOXENV=pylama
-    - TOXENV=pypy-coverage
-    - TOXENV=py27-coverage
+    - JAVA=true  TOXENV=pylama # We build Java once only
```
Comment (iksaif) [inline .travis.yml]: Running the java tests in the pylama env is kind of weird.
What do you think about having a fourth case with JAVA=true and TOXENV being null ?
Also maybe we should make JAVA=false implicit.

**Comment by mcoolive** [comment]
*[.travis.yml, line 33 at r1](https://reviewable.io:443/reviews/criteo/biggraphite/188#-KXGb-wyJeVmZlLitrm0-r1-33:-KXfG4Haqq7qD_2pkTuO:b-pc6f13) ([raw file](https://github.com/criteo/biggraphite/blob/b4943724984b143351b5a12246d0356c0f227f96/.travis.yml#L33)):*
<details><summary><i>Previously, iksaif (Corentin Chary) wrote…</i></summary><blockquote>

Running the java tests in the pylama env is kind of weird.
What do you think about having a fourth case with JAVA=true and TOXENV being null ?
Also maybe we should make JAVA=false implicit.
</blockquote></details>

Currently if JAVA is not specified in the travis matrix, JAVA is tested (so it acts as if true was the default value). I can change that easily.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/188)*

**Comment by mcoolive** [comment]
*[.travis.yml, line 33 at r1](https://reviewable.io:443/reviews/criteo/biggraphite/188#-KXGb-wyJeVmZlLitrm0-r1-33:-KXfJ7kGXgEt_jMUIo6Z:bqmxbtb) ([raw file](https://github.com/criteo/biggraphite/blob/b4943724984b143351b5a12246d0356c0f227f96/.travis.yml#L33)):*
<details><summary><i>Previously, mcoolive (Cyril Martin) wrote…</i></summary><blockquote>

Currently if JAVA is not specified in the travis matrix, JAVA is tested (so it acts as if true was the default value). I can change that easily.
</blockquote></details>

About the mangle Java/Pylama.
I agree mangle Java and Pylama doesn't make sense. First I look for a trick to execute JAVA tests once. I put them with pyjama (and not with other target) because it is the shortest build on travis, so it doesn't affect the build time too much (in fact, the build time increases by 30 sec because we install JDK8 on each env...) 

Separate them seams cleaner. But I don't know how to achieve that in a clean way. What do you mean by "TOXENV being null"? As far I understand, when TOXENV is undefined by travis, tox plays all envs declared in tox.ini. I could wrap the tox command in a "if $PYTHON" or "if ! $ JAVA", but it seams dirty and complex to me.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/188)*

**Comment by iksaif** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[.travis.yml, line 33 at r1](https://reviewable.io:443/reviews/criteo/biggraphite/188#-KXGb-wyJeVmZlLitrm0-r1-33:-KXfUEMNA22FP7o6-H0L:b-y1d5gm) ([raw file](https://github.com/criteo/biggraphite/blob/b4943724984b143351b5a12246d0356c0f227f96/.travis.yml#L33)):*
<details><summary><i>Previously, mcoolive (Cyril Martin) wrote…</i></summary><blockquote>

About the mangle Java/Pylama.
I agree mangle Java and Pylama doesn't make sense. First I look for a trick to execute JAVA tests once. I put them with pyjama (and not with other target) because it is the shortest build on travis, so it doesn't affect the build time too much (in fact, the build time increases by 30 sec because we install JDK8 on each env...) 

Separate them seams cleaner. But I don't know how to achieve that in a clean way. What do you mean by "TOXENV being null"? As far I understand, when TOXENV is undefined by travis, tox plays all envs declared in tox.ini. I could wrap the tox command in a "if $PYTHON" or "if ! $ JAVA", but it seams dirty and complex to me.
</blockquote></details>

What I mean is that this would look cleaner:
- TOXENV=pylama
- TOXENV=pypy-coverage
- TOXENV=py27-coverage
- JAVA=true

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/188)*

**Comment by mcoolive** [comment]
*[.travis.yml, line 33 at r1](https://reviewable.io:443/reviews/criteo/biggraphite/188#-KXGb-wyJeVmZlLitrm0-r1-33:-KXkXSA66nh993ukdpZT:b-896fix) ([raw file](https://github.com/criteo/biggraphite/blob/b4943724984b143351b5a12246d0356c0f227f96/.travis.yml#L33)):*
<details><summary><i>Previously, iksaif (Corentin Chary) wrote…</i></summary><blockquote>

What I mean is that this would look cleaner:
- TOXENV=pylama
- TOXENV=pypy-coverage
- TOXENV=py27-coverage
- JAVA=true


</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/188)*

**Comment by iksaif** [comment]
:lgtm:

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/criteo/biggraphite/188#-:-KXkgz4WolEerTDfKgpX:bnfp4nl)*

**Pull request merged**